Jenkinsfile

```diff
@@ -0,0 +1,151 @@
+#!groovy
+
+pipeline
+{
+  agent none
+
+  stages
+  {
+    stage("check")
+    {
+      agent
+      {
+        docker
+        {
+          image 'dealii/indent'
+        }
+      }
+
+      post { cleanup { cleanWs() } }
+
+      stages
+      {
+        stage("permission")
+        {
+          steps
+          {
+            githubNotify context: 'CI', description: 'need ready to test label and /rebuild',  status: 'PENDING'
+            // For /rebuild to work you need to:
+            // 1) select "issue comment" to be delivered in the github webhook setting
+            // 2) install "GitHub PR Comment Build Plugin" on Jenkins
+            // 3) in project settings select "add property" "Trigger build on pr comment" with
+            //    the phrase ".*/rebuild.*" (without quotes)
+            sh '''
+               wget -q -O - https://api.github.com/repos/dealii/dealii/issues/${CHANGE_ID}/labels | grep 'ready to test' || \
+               { echo "This commit will only be tested when it has the label 'ready to test'. Trigger a rebuild by adding a comment that contains '/rebuild'..."; exit 1; }
+            '''
+            githubNotify context: 'CI', description: 'running tests...',  status: 'PENDING'
+          }
+        }
+
+        stage("indent")
+        {
+          steps
+          {
+            // We can not use 'indent' because we are missing the master branch:
+            // "fatal: ambiguous argument 'master': unknown revision or path"
+            sh '''
+               ./contrib/utilities/indent-all
+            '''
+            sh 'git diff > changes.diff'
+            archiveArtifacts artifacts: 'changes.diff', fingerprint: true
+            sh '''
+               git diff --exit-code || \
+               { echo "Please check indentation, see artifacts at the top right!"; exit 1; }
+            '''
+            githubNotify context: 'indent', description: '',  status: 'SUCCESS'
+          }
+        }
+      }
+    }
+
+    stage('Build and Test')
+    {
+      parallel
+      {
+
+        stage('gcc-serial')
+        {
+          agent
+          {
+            docker
+            {
+              image 'tjhei/candi:v9.0.1-r4'
```

Comment:
You may want to use another tag for the image maybe `latest` or `ci` so that we don't need to update this file every time you update the image.

Reply:
I am not sure this will pull the newest image automatically on all machines and specified the version on purpose. Or does it?
I don't think we will change this image often I hope.